dll/win32/wtsapi32/wtsapi32.c

```diff
@@ -302,23 +321,64 @@ BOOL WINAPI WTSQuerySessionInformationW(
     LPWSTR* Buffer,
     DWORD* BytesReturned)
 {
+    switch (WTSInfoClass)
+    {
+        case WTSUserName:
+        {
+            WCHAR* username;
+            DWORD count = 0;
+
+            GetUserNameW(NULL, &count);
+            if (GetLastError() != ERROR_INSUFFICIENT_BUFFER)
+                return FALSE;
+            if (!(username = heap_alloc(count * sizeof(WCHAR))))
+                return FALSE;
+            GetUserNameW(username, &count);
+            *Buffer = username;
+            *BytesReturned = count * sizeof(WCHAR);
+            return TRUE;
+        }
+
+        case WTSConnectState:
+        {
+            WCHAR* output;
+
+            // Windows always returns 4 bytes even on x64 platform
+            const DWORD size = 4;
+            if (!(output = heap_alloc_zero(size)))
+                return FALSE;
+            *Buffer = output;
+            *BytesReturned = size;
+            return TRUE;
+        }
+
+        case WTSClientProtocolType:
+        {
+            WCHAR* output;
+
+            // USHORT
+            const DWORD size = 2;
+            if (!(output = heap_alloc_zero(size)))
+                return FALSE;
+            *Buffer = output;
+            *BytesReturned = size;
+            return TRUE;
+        }
+
+        default:
+        {
+            if (BytesReturned)
+            {
+                *BytesReturned = 0;
+            }
+            break;
```

Comment:
Should this case `SetLastError(ERROR_INVALID_PARAMETER)` or some variation thereof?

Reply:
No, it's just not implemented yet, so leave old behaviour of simply returning FALSE.